File: sdk/ruby/lib/rocksky/resources/stats.rb
module Rocksky
  module Resources
    # `app.rocksky.stats.*` endpoints.
    class Stats < Base
      # Per-user stats.
      def get_stats(did:)
        query("app.rocksky.stats.getStats", did: did)
      end

      # Year-in-review ("Wrapped") data.
      def get_wrapped(did:, year: nil)
        query("app.rocksky.stats.getWrapped", did: did, year: year)
      end
    end
  end
end

File: sdk/python/src/rocksky/resources/stats.py
"""``app.rocksky.stats.*`` — stats and Wrapped."""

from __future__ import annotations

from typing import Any

from ._base import Resource


class StatsResource(Resource):
    async def get(self, did: str) -> dict[str, Any]:
        """Get aggregate listening stats for an actor."""
        data = await self._transport.query(
            "app.rocksky.stats.getStats", params={"did": did}
        )
        return data or {}

    async def wrapped(self, did: str, *, year: int | None = None) -> dict[str, Any]:
        """Get the Wrapped (year-in-review) payload for an actor.

        Defaults to the most recent completed year on the server side.
        """
        data = await self._transport.query(
            "app.rocksky.stats.getWrapped",
            params={"did": did, "year": year},
        )
        return data or {}
